pkgs/applications/version-management/git-credential-manager/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, autoPatchelfHook, fetchurl, dpkg, makeWrapper, fontconfig, zlib, libcxx, icu, openssl, libX11, libICE, libSM }:
+
+stdenv.mkDerivation rec {
+  pname = "git-credential-manager";
+  version = "2.1.2";
+
+  src = fetchurl {
+    url = "https://github.com/git-ecosystem/git-credential-manager/releases/download/v${version}/gcm-linux_amd64.${version}.deb";
+    sha256 = "sha256-H9VSlY2+MDgUTucttr5mVz+CNrDGhruE1sD5cbKrBqQ=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook dpkg makeWrapper ];
+  sourceRoot = ".";
+  unpackCmd = "dpkg-deb -x $src .";
+
+  buildInputs = [
+    fontconfig
+    zlib
+    libcxx
+    icu
+    openssl
+    libX11
+    libICE
+    libSM
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/bin"
+    cp -R "usr/local/share" "$out/share"
+    chmod -R g-w "$out"
+
+    makeWrapper $out/share/gcm-core/git-credential-manager $out/bin/git-credential-manager \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
+    makeWrapper $out/share/gcm-core/git-credential-manager $out/bin/git-credential-manager-core \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
```

Comment:
Can we patchelf?

Reply:
I can definitely give it a proper shot (still learning all the packaging options and features).

Reply:
Closed in preference of #236850, but otherwise I was able to do it with patchelf. A good learning opportunity. Thank you for your review @SuperSandro2000.